12-watchtower-API.md

```diff
@@ -22,227 +22,320 @@ The scope of this bolt does not include:
-* upon receiving a `cipher` that he does not implement:
-	* MUST reject the appointment. 
-* upon receiving a `transaction_size` too big:
-	* MUST reject the appointment.
-* upon receiving a `transaction_size` too small:
-	* MUST reject the appointment.
-* upon receiving a `transaction_fee` too low:
-	* SHOULD reject the appointment.
-
-* if `qos` was agreed on `wt_init`:
-	* if `op_customer_signature_algorithm` is missing:
-		* MUST reject the appointment.
-	* if `op_customer_signature_algorithm` does not match with one of the supported signing algorithms:
-		* MUST reject the appointment.
-	* if `op_customer_signature` is missing:
-		* MUST reject the appointment.
-	* if `op_customer_public_key` is missing:
-		* MUST reject the appopintment.
-	* if `op_customer_signature` cannot be verified using `op_customer_public_key`:
-		* MUST reject the appopintment.
-
-* if `qos` was not agreed on `wt_init`:
-	* if `op_customer_signature_algorithm` is present:
-		* SHOULD reject the appointment.
-	* if `op_customer_signature` is present:
-		* SHOULD reject the appointment.
-	* if `op_customer_public_key` is present:
-		* SHOULD reject the appopintment.
+* MUST fail the connection  if `locator` does not match any of the previously sent to the WatchTower:
+
+* if `qos` was requested in `appointment`:
+	* MUST fail the connection if `qos_len` is 0.
+	* MUST process `qos_data` according to [Quality of Service data](#quality-of-service-data).
+
+### The `appointment_rejected` message
+
+This message contains information about the rejection of an appointment by the WatchTower.
+
+1. type: ? (`appointment_rejected `)
+2. data:
+   * [`16*byte `:`locator`]
+   * [`u16`: `rcode`]
+   * [`u16`: `reason_len`
+   * [`reason_len*byte`: `reason`]
+
+The sending node:
+
+* MUST receive `appointment` before sending an `appointment_rejected` message.
+* MUST set the `locator` to match the one received in `appointment`.
+* MUST set `rcode` to the rejection code.
+* MAY set and empty `reason` field.
+* MUST set `reason_len` to length of `reason`.
+
+#### Rationale
+
+The `appointment_rejected` message follows the approach taken by the `error` message defined in [BOLT#1](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#the-error-message): error codes are mandatory, whereas reasons are optional and implementation dependant.
+
+## Quality of Service data
+
+`qos_data` is a list where each field specifies they type and associated data of the offered/requested `qos`. The format is defined as follows:
+
+* [`u16`: `qos_type`]
+* [`u16`: `data_len`]
+* [`data_len*byte`: `data`]
+
+So far, only `accountability` is defined.
+
+### `accountability`
+
+The accountability `qos` defines a pair `qos_data` blobs, associated to a pair of messages: The first one is `customer_evidence` and it is provided by the `client` in the `appointment` message. The second one it is `tower_evidence`, and is provided by the WatchTower in the `appointment_accepted` message.
```

Comment:
Maybe "a pair `qos_data` _of_ blobs" (added _of_).

Reply:
or "defines `qos_data` as a pair of blobs".

Reply:
Also "The second one it is" => "The second one is".

Reply:
There are Quality of Service data blobs, so I think `qos_data` blob should be fine

Reply:
@sr-gi plz comment

Reply:
?